chore: Clear Progress section for completed logs & `on_submit` UX

marination opened the pull request:
- Delete `BOM Update Batch` table on 'Completed' log, to save space
- Hide Progress section on 'Completed' log
- Enqueue `on_submit` for 'Update Cost' job, getting leaf boms could take time for huge DBs. Users have to wait for screen to unfreeze.
- Add error handling to `process_boms_cost_level_wise` (Called via cron job and on submit, both in background)
- Extend Log clearing from `Log Settings`(https://github.com/frappe/frappe/pull/17159) to `BOM Update Log`
   <img width="1344" alt="Screenshot 2022-06-21 at 2 07 30 PM" src="https://user-images.githubusercontent.com/25857446/174755877-7c3d29e7-19b5-44c9-ac9c-6f89cdef906e.png">


**Misc:**
- Changed layout of log. `Update Type` and `Status` moved to LHS

marination committed:
chore: Clear Progress section for completed logs & `on_submit` UX

- Delete `BOM Update Batch` table on 'Completed' log, to save space
- Hide Progress section on 'Completed' log
- Enqueue `on_submit` for 'Update Cost' job, getting leaf boms could take time for huge DBs. Users have to wait for screen to unfreeze.
- Add error handling to `process_boms_cost_level_wise` (Called via cron job and on submit, both in background)

ankush commented:
Implement `clear_old_logs` method and users will be able to configure for how long these records need to be kept :p 

https://github.com/frappe/frappe/pull/17159

marination committed:
Merge branch 'develop' into bom-update-log-cleanup-perf

marination committed:
chore: Implement Log clearing interface in BOM Update Log

- Implement Log clearing interface in BOM Update Log
- Add additional info in sidebar: Log clearing only happens for 'Update Cost' type logs
- 'Replace BOM' logs have important info and is used in BOM timeline, so we'll let users decide if they wanna keep or discard it

marination commented:
> Implement `clear_old_logs` method and users will be able to configure for how long these records need to be kept :p
> 
> [frappe/frappe#17159](https://github.com/frappe/frappe/pull/17159)

@ankush done

marination committed:
Merge branch 'develop' into bom-update-log-cleanup-perf
